permission_handler/lib/permission_handler.dart

```diff
@@ -45,7 +45,7 @@ extension PermissionActions on Permission {
   /// been grant access before.
   ///
   /// Returns the new [PermissionStatus].
-  Future<PermissionStatus> request() async {
+  Future<PermissionStatus?> request() async {
```

Comment:
This should also never return `null`, the method should always return a permission status or and exception if something went wrong.